tests/integration/devfile/cmd_devfile_url_test.go

```diff
@@ -29,388 +29,453 @@ var _ = Describe("odo devfile url command tests", func() {
 		helper.CommonAfterEach(commonVar)
 	})
 
-	Context("Listing urls", func() {
-		It("should list url after push using context", func() {
-			// to confirm that --context works we are using a subfolder of the context
-			subFolderContext := filepath.Join(commonVar.Context, helper.RandString(6))
-			helper.MakeDir(subFolderContext)
-			url1 := helper.RandString(5)
-			host := helper.RandString(5) + ".com"
-			url2 := "nodejs-project-3000-" + helper.RandString(5)
-
-			helper.Cmd("odo", "create", "nodejs", "--project", commonVar.Project, "--context", subFolderContext, componentName).ShouldPass()
-
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), subFolderContext)
-			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(subFolderContext, "devfile.yaml"))
-
-			stdout := helper.Cmd("odo", "url", "create", url1, "--port", "3000", "--ingress", "--context", subFolderContext).ShouldFail().Err()
-			Expect(stdout).To(ContainSubstring("host must be provided"))
-
-			stdout = helper.Cmd("odo", "url", "create", url1, "--port", "3000", "--host", host, "--ingress").ShouldFail().Err()
-			Expect(stdout).To(ContainSubstring("the current directory does not represent an odo component"))
-			helper.Cmd("odo", "url", "create", url1, "--port", "3000", "--host", host, "--ingress", "--context", subFolderContext).ShouldPass()
-			helper.Cmd("odo", "url", "create", url2, "--port", "3000", "--host", host, "--ingress", "--context", subFolderContext).ShouldPass()
-			stdout = helper.Cmd("odo", "push", "--context", subFolderContext).ShouldPass().Out()
-			helper.MatchAllInOutput(stdout, []string{url1 + "." + host, url2})
-
-			stdout = helper.Cmd("odo", "url", "list", "--context", subFolderContext).ShouldPass().Out()
-			helper.MatchAllInOutput(stdout, []string{url1, url2, "Pushed", "false", "ingress"})
-		})
-
-		It("should list ingress url with appropriate state", func() {
-			url1 := helper.RandString(5)
-			url2 := helper.RandString(5)
-			host := helper.RandString(5) + ".com"
+	When("creating a Nodejs component", func() {
+		url1 := helper.RandString(5)
```

Comment:
Please declare the variables as close as possible to their usage. Here, I think that url1 should be declared into the block `When("creating ingress url1 with port flag and doing odo push", func() {`

Reply:
I did that as it was getting used at multiple test case, but yes, I will change this in the next push.